recording/lib/edge_timestamp_saver.ex

```diff
@@ -64,11 +69,26 @@ defmodule Membrane.RTC.Engine.Endpoint.Recording.EdgeTimestampSaver do
     )
 
     {[buffer: {:output, buffer}],
-     %{
-       state
-       | first_buffer_timestamp: timestamp,
-         last_buffer_timestamp: timestamp
-     }}
+     %{state | first_buffer_timestamp: timestamp, last_buffer_timestamp: timestamp}}
+  end
+
+  @impl true
+  def handle_buffer(
+        _pad,
+        %Buffer{} = buffer,
+        ctx,
+        %{first_buffer_timestamp: first_buffer_timestamp} = state
+      )
+      when in_order?(first_buffer_timestamp - buffer.metadata.rtp.timestamp) do
+    timestamp = buffer.metadata.rtp.timestamp
+
+    Recording.Reporter.start_timestamp(
+      state.reporter,
+      track_id(ctx),
+      timestamp
+    )
+
+    {[buffer: {:output, buffer}], %{state | first_buffer_timestamp: timestamp}}
```

Comment:
Here, we don't handle the logic related to the counter. I'm not sure if this is a big issue, as this could happen only at the beginning of the stream, but I want to make sure that you know it.